examples/hybrid/gravitywave_parameterization/gw_test_3d.jl

```diff
@@ -0,0 +1,99 @@
+using NCDatasets
+using Dates
+using Interpolations
+using Statistics
+using Plots
+include("gravity_wave_parameterization.jl")
+const FT = Float64
+
+# test Figure 8 of the Alexander and Dunkerton (1999) paper:
+# https://journals.ametsoc.org/view/journals/atsc/56/24/1520-0469_1999_056_4167_aspomf_2.0.co_2.xml?tab_body=pdf
+
+face_z = 0:1e3:0.47e5
+center_z = 0.5.*(face_z[1:end-1].+face_z[2:end])
+
+
+# compute the source parameters
+params = gravity_wave_cache(FT; Bm = 0.4, cmax = 150, kwv = 2π/100e3 )
+source_level = argmin( abs.(center_z .- params.gw_source_height) )
+
+# ERA5 data 1973 Jan
+ds = NCDataset("./era5-monthly.nc")
+
+lon = ds["longitude"][:] 
+lat = ds["latitude"][:] 
+lev = ds["level"][:] .* 100
+time = ds["time"][:]
+
+gZ = ds["z"][:] 
+T = ds["t"][:] 
+u = ds["u"][:]
```

Comment:
Data leak!
```suggestion
NCDataset("./era5-monthly.nc") do ds
    lon = ds["longitude"][:]
    lat = ds["latitude"][:]
    lev = ds["level"][:] .* 100
    time = ds["time"][:]

    gZ = ds["z"][:]
    T = ds["t"][:]
    u = ds["u"][:]
end
```

if this doesn't work, this will:

```suggestion
nt = NCDataset("./era5-monthly.nc") do ds
    lon = ds["longitude"][:]
    lat = ds["latitude"][:]
    lev = ds["level"][:] .* 100
    time = ds["time"][:]
    gZ = ds["z"][:]
    T = ds["t"][:]
    u = ds["u"][:]
    (; lon, lat, lev, time, gZ, T, u)
end
(; lon, lat, lev, time, gZ, T, u) = nt
```

Reply:
and use the `do` syntax for the other `NCDataset` calls, please!